src/hostapi/coreaudio/pa_mac_core_blocking.c

```diff
@@ -62,7 +62,14 @@
 
 #include "pa_mac_core_blocking.h"
 #include "pa_mac_core_internal.h"
-#include <assert.h>
+
+#ifdef NDEBUG
+# undef assert
+# define assert(VAL) do { (void)sizeof(VAL); } while(0)
```

Comment:
Thanks for this. The use of `sizeof` here risks not evaluating side effects of VAL. (Of course well written code would have no side effects in an assert or debug print, but we should not assume that.) Did you try the following instead:

` # define assert(VAL) do { (void)(VAL); } while(0)`

If not, please try it, it's preferable if it works. Same for the other DBUG macros.

Reply:
I have tried `(void)(VAL)`, but I went for sizeof because I got `expression result unused` errors with `VDBUG` and `VVDBUG`:

```
src/hostapi/coreaudio/pa_mac_core.c:357:14: error: expression result unused [-Werror,-Wunused-value]
    VDBUG( ( "Found %ld device(s).\n", auhalHostApi->devCount ) );
             ^~~~~~~~~~~~~~~~~~~~~~~~
src/hostapi/coreaudio/pa_mac_core_utilities.h:101:33: note: expanded from macro 'VDBUG'
# define VDBUG(MSG) do { (void)(MSG); } while(0)
                                ^~~
```
```
src/hostapi/coreaudio/pa_mac_core.c:1193:13: error: expression result unused [-Werror,-Wunused-value]
    VVDBUG(("OpenAndSetupOneAudioUnit(): in chan=%d, in fmt=%ld, out chan=%d, out fmt=%ld, requestedFramesPerBuffer=%ld\n",
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/hostapi/coreaudio/pa_mac_core_utilities.h:111:34: note: expanded from macro 'VVDBUG'
# define VVDBUG(MSG) do { (void)(MSG); } while(0)
                                 ^~~

```
```
src/hostapi/coreaudio/pa_mac_core.c:1194:63: error: expression result unused [-Werror,-Wunused-value]
            inStreamParams  ? inStreamParams->channelCount  : -1,
                              ~~~~~~~~~~~~~~                  ^~
src/hostapi/coreaudio/pa_mac_core_utilities.h:111:34: note: expanded from macro 'VVDBUG'
# define VVDBUG(MSG) do { (void)(MSG); } while(0)
                                 ^~~
```

Reply:
Alright. In addition to the fact that you've provided evidence that `sizeof` is needed to provide a valid fix for the compiler warnings, I thought about this some more and realised that I was wrong to argue for removing `sizeof`: The old code did not evaluate `VAL` in the no-debug case, It used an empty define, e.g,

`# define DBUG(MSG)`

So your patch that uses `sizeof` is entirely equivalent to the current behavior of those macros. I approve.